### PR TITLE
chore: add mlsVerificationStatus to ZMConversation

### PR DESCRIPTION
**Pull request by caldrian**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

On `develop` as well as on `feat/e2ei` a zmessaging model revision 113 have been created. This PR merges both changes.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-data-model/Source/Model/Conversation/ZMConversation+MLS.swift [hunk 1]
```diff
@@ -133,6 +136,37 @@ extension ZMConversation {
...
+
+            return status
+        }
+
+        set {
+            willChangeValue(forKey: Self.mlsVerificationStatusKey)
+            primitiveMlsVerificationStatus = newValue.map { NSNumber(value: $0.rawValue) }
+            didChangeValue(forKey: Self.mlsVerificationStatusKey)
+        }
+    }
```
Comment (caldrian) [inline wire-ios-data-model/Source/Model/Conversation/ZMConversation+MLS.swift:168]: I wonder if this property could be non-optional with the raw value of `.notVerified`.

```suggestion
    @NSManaged private var primitiveMlsVerificationStatus: NSNumber

    /// The mls verification status.
    ///
    /// If this conversation is an mls group (which it should be if the
    /// `messageProtocol` is `mls`), then this identifier should exist.

    public var mlsVerificationStatus: MLSVerificationStatus {
        get {
            willAccessValue(forKey: Self.mlsVerificationStatusKey)
            let value = primitiveMlsVerificationStatus.int16Value
            didAccessValue(forKey: Self.mlsVerificationStatusKey)

            guard let status = MLSVerificationStatus(rawValue: value) else {
                fatalError("failed to init MLSVerificationStatus from rawValue: \(value)")
            }

            return status
        }
        set {
            willChangeValue(forKey: Self.mlsVerificationStatusKey)
            primitiveMlsVerificationStatus = .init(value: newValue.rawValue)
            didChangeValue(forKey: Self.mlsVerificationStatusKey)
        }
    }
```

File: wire-ios-data-model/Resources/zmessaging.xcdatamodeld/zmessaging2.112.0.xcdatamodel/contents [hunk 2]
```diff
@@ -320,7 +320,7 @@
         <attribute name="remoteIdentifier_data" optional="YES" attributeType="Binary"/>
         <attribute name="richProfile" optional="YES" attributeType="Binary"/>
         <attribute name="serviceIdentifier" optional="YES" attributeType="String"/>
-        <attribute name="supportedProtocols" optional="YES" attributeType="Transformable"/>
+        <attribute name="supportedProtocols" optional="YES" attributeType="Transformable" valueTransformerName="ExtendedSecureUnarchiveFromData"/>
```
Comment (markusfassbender) [inline wire-ios-data-model/Resources/zmessaging.xcdatamodeld/zmessaging2.112.0.xcdatamodel/contents]: Do we need to add it here in xcdatamodel `112` if I see the same changes in `113`?

File: wire-ios-data-model/Source/Model/Conversation/ZMConversation+MLS.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios-data-model/Source/Model/Conversation/ZMConversation+MLS.swift:168]: I'm fine to have this property as non-optional. For `e2ei` this shouldn't break anything. @johnxnguyen What do you think? I know we have some other mls properties that are optional.

File: wire-ios-data-model/Resources/zmessaging.xcdatamodeld/zmessaging2.112.0.xcdatamodel/contents (hunk 2 above)
Comment (caldrian) [reply wire-ios-data-model/Resources/zmessaging.xcdatamodeld/zmessaging2.112.0.xcdatamodel/contents]: Reverted. 👍

**Review by markusfassbender — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**